### PR TITLE
ZCS-12405: WCAG | Classic UI | Medium Complexity | Aria issues Part - 2

### PR DESCRIPTION
**Pull request by miteshsavani810**
**4.1.2_text alternative is not provided for the button.docx**

> Added `aria-label` to each `div` which having role='button' attribute.

**4.1.2_name role issue-Mail.docx**

> There is no popup set with `username` field. Popup has been associated with next element `link` field. All requires attributes are already set to that `link` element. However We set aria-disabled='true' attribute to `username` field to not announce it as "Clickable".

**1.3.1_table issue-Mail.docx**
**1.3.1_list table issue-New Mail.docx**

>  Added role='presentation' attribute to each `table` element and tr/td elements which are part of popup block.

**Notes:** Additionally, Added aria-expanded='false' attribute to each drop down button and when popup is open then app will set aria-expanded as 'true' and when popup get closed it will set aria-expanded as 'false'.

See the changes of zm-ajax https://github.com/Zimbra/zm-ajax/pull/103

### CODE REVIEW

File: WebRoot/WEB-INF/tags/calendar/dayAppt.tag [hunk 1]
```diff
@@ -35,7 +35,7 @@
 <c:set var="subject" value="${empty appt.name ? noSubject : appt.name}"/>
 <app:calendarUrl appt="${appt}" var="apptUrl"/>
 <c:if test="${selected}">
-    <table width="100%" border="0" style="height:100%;" cellspacing="0" cellpadding="0">
+    <table role='presentation' width="100%" border="0" style="height:100%;" cellspacing="0" cellpadding="0">
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/WEB-INF/tags/calendar/dayAppt.tag:38]: role="presentation"

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js [hunk 2]
```diff
@@ -483,7 +483,8 @@ function() {
 	var subs = {
 		id: 			this._htmlElId,
 		alphabet: 		alphabet,
-		numLetters: 	alphabet.length
+		numLetters: 	alphabet.length,
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js:486]: indentation

File: WebRoot/js/zimbraMail/share/view/dialog/ZmFolderNotifyDialog.js [hunk 3]
```diff
@@ -187,6 +187,7 @@ function(organizer) {
 	if (visible) {
 		AjxDispatcher.require("Share");
 		var table = document.createElement("TABLE");
+        table.setAttribute('role', 'presentation');
```
Comment (HimanshuSojitraSynacor) [inline WebRoot/js/zimbraMail/share/view/dialog/ZmFolderNotifyDialog.js:190]: indentation

File: WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js [hunk 4]
```diff
@@ -2415,7 +2416,7 @@ ZmEditContactViewOther.__DwtButton_popup = function() {
 
 ZmEditContactViewOther.prototype._createSelect = function() {
 	var id = [this.getHTMLElId(),"select"].join("_");
-	var select = new DwtComboBox({parent:this,inputParams:{size:14},id:id});
+	var select = new DwtComboBox({parent:this,inputParams:{size:14},id:id, ariaLabel: 'Test'});
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js:2419]: Test?

File: WebRoot/js/zimbraMail/core/ZmZimbraMail.js [hunk 5]
```diff
@@ -2666,6 +2667,7 @@ ZmZimbraMail.prototype.setUserInfo = function() {
 		var el = this._userNameField.getHtmlElement();
         el.innerHTML =  AjxStringUtil.htmlEncode(AjxStringUtil.clipByLength(username, 24));
 		el.setAttribute('aria-label', username);
+		el.setAttribute('aria-disabled', true); // Prevent SR to annouch as clickable.
```
Comment (silentsakky) [inline WebRoot/js/zimbraMail/core/ZmZimbraMail.js:2670]:
```suggestion
		el.setAttribute('aria-disabled', true); // Prevent SR to announce as clickable.
```

Comment (silentsakky) [inline WebRoot/js/zimbraMail/core/ZmZimbraMail.js:2670]: isn't removing aria attributes the same as aria-disabled?

File: WebRoot/WEB-INF/tags/calendar/dayAppt.tag (hunk 1 above)
Comment (miteshsavani810) [reply WebRoot/WEB-INF/tags/calendar/dayAppt.tag]: Fixed

File: WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js (hunk 2 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/abook/view/ZmContactsBaseView.js]: Fixed

File: WebRoot/js/zimbraMail/share/view/ZmSearchToolBar.js [hunk 6]
```diff
@@ -377,6 +377,7 @@ ZmMainSearchToolBar.prototype._initialize = function() {
         id:         ZmId.getButtonId(ZmId.SEARCH, ZmId.SEARCH_SEARCH)
     });
     button.setImage("Search2");
+	button._textSet(ZmMsg.searchTooltip);
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/view/ZmSearchToolBar.js]: Fixed

File: WebRoot/js/zimbraMail/share/view/dialog/ZmFolderNotifyDialog.js (hunk 3 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/share/view/dialog/ZmFolderNotifyDialog.js]: Fixed

File: WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js (hunk 4 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/abook/view/ZmEditContactView.js]: Removed.

File: WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js [hunk 7]
```diff
@@ -1092,6 +1092,7 @@ function(viewId) {
 	var printButton = toolbar.getButton(ZmOperation.PRINT_CALENDAR);
 	if (printButton) {
 		printButton.setToolTipContent(ZmMsg.printCalendar);
+		printButton._textSet(ZmMsg.printCalendar);
```
[the remark this answers is not in the document]
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/calendar/controller/ZmCalViewController.js]: Updated method name as `setAriaLabel`

File: WebRoot/js/zimbraMail/core/ZmZimbraMail.js (hunk 5 above)
Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/core/ZmZimbraMail.js]: Done.

Comment (miteshsavani810) [reply WebRoot/js/zimbraMail/core/ZmZimbraMail.js]: Ignoring the comment as per our offline discussion.

**Review by HimanshuSojitraSynacor — APPROVED** [review]

**Review by silentsakky — APPROVED** [review]

**Review by lavegupta — APPROVED** [review]

**Pull request merged**